lib/view_component/base.rb

```diff
@@ -143,6 +143,10 @@ def request
     class << self
       attr_accessor :source_location
 
+      def all_templates_inlined?
+        @all_templates_inlined ||= inline_calls.any? && templates.empty?
```

Comment:
Do we need this predicate for non-testing purposes? If not, can we remove it, or at least take it out of the rendering path?

Reply:
You're absolutely right - no need for this at all, will remove!